fegrow/mlp.py

```diff
@@ -0,0 +1,175 @@
+
+        Returns:
+            bool: True if compatible, False otherwise.
+        """
+        mol_elements = set(atom.symbol for atom in molecule.atoms)
+        if not mol_elements.issubset(cls.allowed_elements):
+            return False
+        # Work out the total formal charge
+        rdkit_mol = molecule.to_rdkit()
+        total_charge = sum(atom.GetFormalCharge() for atom in rdkit_mol.GetAtoms())
+        if not cls.allow_charged and total_charge != 0:
+            return False
+        return True
+
+
+class ANI2X(_MLForceField):
+    name = "ani2x"
+    allowed_elements = frozenset(["H", "C", "N", "O", "S", "F", "Cl"])
+    allow_charged = False
+
+    @classmethod
+    def get_potential(cls) -> OMMMLPotential:
+        """Get the ANI2x potential."""
+        return OMMMLPotential(cls.name)
+
+
+class _MACE_BASE(_MLForceField):
+    """Base class for MACE force fields."""
+
+    allowed_elements = frozenset(["C", "H", "N", "O", "S", "F", "Cl", "Br"])
+    allow_charged = False
+
+    @classmethod
+    def get_potential(cls) -> OMMMLPotential:
+        """Get the MACE potential."""
+
+        cls._check_available()
+
+        print(
+            "MACE models are distributed under the ASL which"
+            "does not permit commercial use."
+        )
+
+        return OMMMLPotential(cls.name)
+
+    @staticmethod
+    def _check_available() -> None:
+        """Check if the MACE force field is available."""
+        _check_mace_installed()
+
+
+class MACEOFF23Small(_MACE_BASE):
+    """MACE-OFF 23 Small force field."""
+
+    name = "mace-off23-small"
+
+
+class MACEOFF23Medium(_MACE_BASE):
+    """MACE-OFF 23 Medium force field."""
+
+    name = "mace-off23-medium"
+
+
+class MACEOFF23Large(_MACE_BASE):
+    """MACE-OFF 23 Large force field."""
+
+    name = "mace-off23-large"
+
+
+class EGRET1(_MLForceField):
+    """EGRET-1 force field."""
+
+    name = "egret-1"
+    allowed_elements = frozenset(["C", "H", "N", "O", "S", "F", "Cl", "Br"])
+    allow_charged = False
+
+    @classmethod
+    def get_potential(cls) -> OMMMLPotential:
+        """Get the EGRET-1 potential."""
+
+        cls._check_available()
+
+        tmp_file = tempfile.NamedTemporaryFile(suffix=".model", delete=False)
+        tmp_file.close()  # Close so urllib can write to it
+        print(f"Downloading Egret-1 model from {_EGRET1_MODEL_URL}")
+        urllib.request.urlretrieve(_EGRET1_MODEL_URL, filename=tmp_file.name)
+
+        # Register file for deletion at program exit
+        atexit.register(
+            lambda: os.remove(tmp_file.name) if os.path.exists(tmp_file.name) else None
+        )
+
+        return OMMMLPotential("mace", modelPath=tmp_file.name)
+
+    @staticmethod
+    def _check_available() -> None:
+        """Check if the EGRET-1 MLP is available."""
+        _check_mace_installed()
+
+
+_MLFF_NAME_TO_CLASS: dict[AVAILABLE_ML_FORCE_FIELDS, type[_MLForceField]] = {
```

Comment:
similarly, nice functionality, no need for _, we'd want other to use this interface if they want to, also maybe `MLFF_CLASS`, so you can say MLFF_CLASS['ani2x']